rasa/core/evaluation/stats.py

```diff
@@ -0,0 +1,112 @@
+import os
+from typing import Dict, Text, Union, Any, List, Tuple, TypedDict
+from pathlib import Path
+import json
+import numpy as np
+
+
+class Stats(TypedDict):
+    """A TypedDict for stats."""
+
+    n: int
+    mean: float
+    median: float
+    min: int
+    max: int
+
+
+def load_extracted_markers_json_file(path: Union[Text, Path]) -> List:
+    """Reads a json marker file.
+
+    Args:
+        path: path to a json file.
+    """
+    path = os.path.abspath(path)
+    with open(path) as json_file:
+        extracted_markers = json.load(json_file)
+        return extracted_markers
+
+
+def compute_summary_stats(data_points: Union[List, np.ndarray]) -> Stats:
+    """Computes summary statistics for a given array.
+
+    Computes size, mean, median, min, and max.
+    If size is == 0 returns np.nan for mean, median.
+
+    Args:
+        data_points: can be a numpy array or a list of numbers.
+    """
+    if np.size(data_points) > 0:
+        stats: Stats = {
+            "n": int(np.size(data_points)),
+            "mean": float(np.mean(data_points)),
+            "median": float(np.median(data_points)),
+            "min": int(np.min(data_points)),
+            "max": int(np.max(data_points)),
+        }
+        return stats
```

Comment:
@usc-m let me know if I've used the TypedDict correctly.

Reply:
Only thing I'd say is the name is a bit generic and we could end up with namespace collisions or some confusion if this structure is referenced elsewhere